src/psyclone/psyir/nodes/schedule.py

```diff
@@ -53,6 +54,11 @@ class Schedule(Node):
     '''
     def __init__(self, children=None, parent=None):
         Node.__init__(self, children=children, parent=parent)
+        # TODO #645 remove this check that we don't already have a symbol
+        # table (only currently required because InvokeSchedule creates its
+        # own symbol table *before* calling this constructor).
+        if not (hasattr(self, "_symbol_table") and self._symbol_table):
```

Comment:
Probably a false error but I've added it just in case.
pylint:
E: 60,51: Access to member '_symbol_table' before its definition line 61 (access-member-before-definition)

Reply:
Yes, pylint doesn't pick up the fact that we use `hasattr` before we attempt to access it.